src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,133 @@
+# Consider the ODE dxₜ/dt = f(xₜ,p). We want to ramp one parameter of this ODE.
+"""
+    RateConfig
+
+Time-dependent forcing protocol ``p(t)`` describing the evolution of a parameter over a
+time interval `[section_start, section_end]`.
+Used to construct a non-autonomous `RateSystem`.
+
+## Fields
+- `pfunc`: function ``p(t)`` from ``ℝ → ℝ`` describing the parameter time dependence
+- `section_start`: start of the time interval over which `pfunc` is considered
+- `section_end`: end of the time interval over which `pfunc` is considered
+
+## Description
+The `RateConfig` type allows to specify the functional form of a parametric
+forcing over a time interval. This forcing protocol can then be applied to the parameter
+of a dynamical system using the `RateSystem` constructor, which also allows to modify
+the rate and magnitude of the forcing.
+"""
+mutable struct RateConfig
+    pfunc::Function
+    section_start::Float64
+    section_end::Float64
+end
+
+"""
+    RateSystem
+ 
+"""
+struct RateSystem
+    system
+    forcing
+    pidx
+end
+
+"""
+    RateSystem(sys::ContinuousTimeDynamicalSystem, rc::RateConfig, pidx; kwargs...)
+
+Creates a `RateSystem` type from an autonomous dynamical system `sys` and a time-dependent
+parametric forcing protocol of `RateConfig` type.
+
+## Keyword arguments
+- `forcing_start = RateConfig.section_start`: Time when parameter shift starts (before this, the resulting system will be autonomous)
+- `forcing_length = RateConfig.section_end - RateConfig.section_start`: Time-interval over which RateConfig.pfunc([RateConfig.section_start, RateConfig.section_end]) is spread out (for window_length > RateConfig.section_end - RateConfig.section_start) or squeezed into (for window_length < RateConfig.section_end - RateConfig.section_start)
+- `forcing_scale = 1.0`: Amplitude of the ramping. The ramping is then automatically rescaled 
+- `t0 = 0.0`: Initial time of the resulting non-autonomous system (relevant to later compute trajectories)
+
+## Description
+The returned `RateSystem.system` is is 
```

Comment:
You shouldn't be referring to `RateSystem.system` here. It isn't valid syntax either, as `RateSyste` is a type and does not have a field `system`. Regardless, just say `rs = RateSystem(...)` and refer to `rs` exclusively. It is a dynamical system because we extended the proper API.